### ISSUE TITLE
[a11y]: Modal - "content is not within a landmark element"

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#12773]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

11.17

### React version

1.17

### Automated testing tool and ruleset

EAAC latest version

### Description

EAAC flags errors for modals due to the `<span tabindex="0" role="link" class="cds--visually-hidden">Focus sentinel</span>` nodes that exist outside of the `role="dialog"`.

<img width="1287" alt="Screenshot 2022-12-05 at 15 48 23" src="https://user-images.githubusercontent.com/69599/205568584-599703ca-f2bb-446a-9b00-9717a01a6615.png">

Probably they should be inside the dialog.

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=&id=components-modal--default&viewMode=story

### Steps to reproduce

Just run EAAC on above link.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for reporting this! 

Technically this reports as a violation, as you've found. After discussing this and the other components that use focus sentinels (like tooltip, https://github.com/carbon-design-system/carbon/issues/12047#issuecomment-1255076849) with IBMa we agreed this is more-so a false positive reported by the tool since they're visually hidden.

**Comment by wkeese** [comment]
OK, being visually hidden seems irrelevant though.  Landmark roles are (mainly) for blind people, they don't care what is/isn't visually hidden.

Having said that, apparently you use the sentinels to prevent the user from tabbing out of the dialog back onto the main page, and as soon as the user navigates to a sentinel you shift focus back into the dialog.  So hopefully that's quick enough that the screen readers don't announce anything related to the sentinel.
